### DISCUSSION TITLE
[Contract Verification] Details on how the verification process works

### DISCUSSION THREAD

**Discussion by wighawag** [zkSync-Community-Hub/zksync-developers#467] [Contract Verification] [answered]
### Environment

Testnet

### Issue Type

- [ ] Plugin
- [ ] Block Explorer

### Contract Address

Not relevant

### Compiler Type

Multi-part contract

### zkSolc Version

latest

### Solc Version

latest

### Contract Name

Not relevant

### Contract Code

any

### Constructor Arguments

any

### Additional Details

I have my own verification setup and would like to support zksync

Unfortunately your doc redirect to tools and do not offer an explanation of how the process can work

https://docs.zksync.io/build/tooling/hardhat/hardhat-zksync-verify.html#commands

I noticed that the artifact generated by zksolc provide a metadata output that differs from normal solc

What I am supposed to send to sourcify / etherscan / blockscout  api ?

Any link to your source code would be great

Thanks

**Comment by bxpana** [comment]
> do not offer an explanation of how the process can work

Do you mean show exactly how the contract is verified behind the scenes of the tools?

> What I am supposed to send to sourcify / etherscan / blockscout api ?

Right now my understanding is that using the hardhat verify tool only supports verifying contracts on https://explorer.zksync.io/, but we have plans to extend that in the future to support https://era.zksync.network/ (Etherscan version).

**Reply by wighawag** [reply]
> Do you mean show exactly how the contract is verified behind the scenes of the tools?

yes exactly

**Reply by bxpana** [reply]
Here's the repo if that helps https://github.com/matter-labs/hardhat-zksync/tree/main/packages/hardhat-zksync-verify

**Reply by bxpana** [reply]
Did this get you what you needed?

**Reply by wighawag** [reply]
I guess I can work with that, but ideally the doc should explain all what is needed without needing to look at specific framework integration code

**Reply by dutterbutter** [reply]
@wighawag there is also this doc outlining contract verification: https://docs.zksync.io/build/tutorials/how-to/verify-contracts.html

**Reply by wighawag** [reply]
This is just a doc on how to use a tool to verify. it does not explain how your zksync explorer api work and what data format, etc,,... it expects

**Comment by gepng** [comment, marked as answer]
try this: https://block-explorer-api.mainnet.zksync.io/docs#/Contract%20API/ApiController_verifyContractSourceCode

Its quite similar to normal etherscan api with some differences

**Reply by bxpana** [reply]
@wighawag did this help you for what you were looking for? ☝️

**Reply by wighawag** [reply]
Yes should help, but how is the order of the library defined ?

**Reply by gepng** [reply]
if you submit everything as standart-json-input you should not need the library specifications at all. Or at least I dont when im doing mine. @wighawag
